DigitalLearningSolutions.Data/Services/CustomPromptsService.cs

```diff
@@ -157,16 +172,54 @@ public string GetPromptNameForCentreAndPromptNumber(int centreId, int promptNumb
             return customPromptsDataService.GetPromptNameForCentreAndPromptNumber(centreId, promptNumber);
         }
 
-        public CourseCustomPrompts? GetCustomPromptsForCourse(int customisationId, int centreId, int categoryId)
+        public CourseCustomPrompts? GetCustomPromptsForCourse(
+            int customisationId,
+            int centreId,
+            int categoryId = 0,
+            bool includeArchived = false
+        )
         {
-            var result = customPromptsDataService.GetCourseCustomPrompts(customisationId, centreId, categoryId);
-            return result == null
-                ? null
-                : new CourseCustomPrompts(
-                    customisationId,
-                    centreId,
-                    PopulateCustomPromptListFromCourseCustomPromptsResult(result)
-                );
+            var result = GetCourseCustomPromptsResultForCourse(customisationId, centreId, categoryId, includeArchived);
+            if (result == null)
+            {
+                return null;
+            }
+
+            return new CourseCustomPrompts(
+                customisationId,
+                centreId,
+                PopulateCustomPromptListFromCourseCustomPromptsResult(result)
+            );
+        }
+
+        public List<CustomPromptWithAnswer> GetCustomPromptsWithAnswersForCourse(
+            DelegateCourseInfo delegateCourseInfo,
+            int customisationId,
+            int centreId,
+            int categoryId = 0,
+            bool includeArchived = false
+        )
+        {
+            var result = GetCourseCustomPromptsResultForCourse(customisationId, centreId, categoryId, includeArchived);
+
+            return PopulateCustomPromptWithAnswerListFromCourseCustomPromptsResult(result, delegateCourseInfo);
+        }
+
+        private CourseCustomPromptsResult? GetCourseCustomPromptsResultForCourse(
+            int customisationId,
+            int centreId,
+            int categoryId,
+            bool includeArchived

```

Comment:
Just marking where this is so I don't have to find it again. This should be removed as we add it back to SQL